### DISCUSSION TITLE
《Rest 参数与 Spread 语法》中可能存在的问题

### DISCUSSION THREAD

**Discussion by smallpigz** [javascript-tutorial/zh.javascript.info#972] [Q&A] [answered]
在《Rest 参数与 Spread 语法》章节中介绍说明

![Issues1](https://user-images.githubusercontent.com/41250661/142994424-a3b91856-2cfc-4f69-ae45-9b568600637e.png)

在该叙述下面的例子中用到 Spread 语法复制对象

<img width="725" alt="Issues2" src="https://user-images.githubusercontent.com/41250661/142994425-fa0ad6c0-6d4e-44df-89ef-9709cc045867.png">

在前面的章节《Iterable **object（可迭代对象）》中介绍了*普通对象不是迭代对象*

所以对红框中的描述有点疑惑，找了MDN。

![Issues3](https://user-images.githubusercontent.com/41250661/142994844-171b463e-7d1a-4177-a245-a9c16b29403f.png)

大佬们，这个地方是否存在歧义😂

**Comment by songhn233** [comment, marked as answer]
这两种说法是不冲突的。

第一种情况是 [在数组中使用 spread 语法](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#spread_in_array_literals)，第二种是 [在对象字面量中使用 spread 语法](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#spread_in_array_literals)。

既然提到 MDN 文档的话，其实文档中 [也有说明](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#only_for_iterables)。

![image](https://user-images.githubusercontent.com/47357585/143077767-e42432b4-bd7d-4265-92d0-62b2d3fd71df.png)

本教程该章节提到的「Spread 语法只适用于可迭代对象」是在 `Array.from` 和 `[...obj]` 的数组上下文的提到的，所以没有问题。

**Reply by smallpigz** [reply]
明白了，谢谢解答😄
